trace-sdk/src/test/java/io/bitrise/trace/data/collector/view/ActivityStateDataListenerTest.java

```diff
@@ -0,0 +1,47 @@
+package io.bitrise.trace.data.collector.view;
+
+import android.app.Activity;
+import android.content.Context;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+
+/**
+ * Unit tests for {@link ActivityStateDataListener}.
+ */
+public class ActivityStateDataListenerTest {
+
+    final Context mockContext = Mockito.mock(Context.class);
+    final ActivityStateDataListener listener = new ActivityStateDataListener(mockContext);
+    final Activity mockActivity = Mockito.mock(Activity.class);
+
+    @Test
+    public void onActivityCreated_notActive() {
```

Comment:
not Active tests already tested with activityMap_ShouldNotContainActivityWhenCollectingStopped in ActivityStateDataListenerInstrumentedTest

Reply:
I think if you look at the coverage report on main for that it's displaying those rows as red - https://codecov.io/gh/bitrise-io/trace-android-sdk/src/main/trace-sdk/src/main/java/io/bitrise/trace/data/collector/view/ActivityStateDataListener.java - so i think theses tests still are valid.

Reply:
I think I got the reason. So this assertion is redundant, but that line is not covered, because stopCollecting also unregisters the listener, so the onStop method of the listener will be not called, because the traceActivityLifecycleTracker will not forward it.
So I was considering if those isActive checks are needed or not, because there is no use case when it is not active, but registered. From tech perspective, we cannot foretell if the usage of those method changes in the future (and how), so I believe, for safety reasons we should keep the checks, and therefore the tests are needed too.
On the other hand, a really similar case you mentioned [here](https://github.com/bitrise-io/trace-android-sdk/pull/199#discussion_r633488933), because of current usage the (!isApplicationStart) check is redundant, but applying the same principle, I think we should keep that check there too. Thoughts?

Reply:
Thanks for the comment Richard. I think I agree with what you're saying, this class (and the other) would maybe worth a bit of a refactor bearing that information in mind. My intention for this ticket wasn't to try and refactor any of the existing code (acceptance criteria is only to increase current code coverage). Let's add this to the android catchup agenda on Monday to discuss how we can improve this going forward? Would that work for you?

Reply:
So you removed the check in one place, but kept in the other and added test cases for it. This is not about refactoring, but inconsistency in your PR.

Reply:
I don't think I've removed any code - i definitely didn't intend to remove any checks. Can you point to example of this so I can put it back? There are some line deletions because I moved the collection creation to class visibility, instead of each test creating it, but that should be the only change to existing tests here.